src/overlays/actors/ovl_En_Skb/z_en_skb.c

```diff
@@ -66,55 +74,499 @@ static ColliderJntSphInit D_80AFE068 = {
+        func_80AFCD60(this);
+    }
+}
+
+void EnSkb_SetupStunned(EnSkb* this) {
+    if (this->actor.bgCheckFlags & 1) {
+        this->actor.speedXZ = 0.0f;
+    }
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_JR_FREEZE);
+    this->unk_281 = 0;
+    this->unk_280 = 6;
+    EnSkb_SetupAction(this, func_80AFD59C);
+}
+
+void func_80AFD59C(EnSkb* this, GlobalContext* globalCtx) {
+
+    if ((this->actor.bgCheckFlags & 2) != 0) {
+        this->actor.speedXZ = 0.0f;
+    }
+    if ((this->actor.bgCheckFlags & 1) != 0) {
+        if (this->actor.speedXZ < 0.0f) {
+            this->actor.speedXZ += 0.05f;
+        }
+    }
+    if ((this->actor.colorFilterTimer == 0) && ((this->actor.bgCheckFlags & 1) != 0)) {
+        if (this->actor.colChkInfo.health == 0) {
+            func_80AFD7B4(this, globalCtx);
+            return;
+        }
+        func_80AFCD60(this);
+    }
+}
+
+void func_80AFD644(EnSkb* this) {
+    Animation_MorphToPlayOnce(&this->skelAnime, &D_06000D98, -4.0f);
+    if (this->actor.bgCheckFlags & 1) {
+        this->actor.speedXZ = -4.0f;
+    }
+    this->actor.world.rot.y = this->actor.yawTowardsPlayer;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_STALKID_DAMAGE);
+    this->unk_280 = 2;
+    EnSkb_SetupAction(this, func_80AFD6CC);
+}
+
+void func_80AFD6CC(EnSkb* this, GlobalContext* globalCtx) {
+    // this cast is likely not real, but allows for a match
+    u8* new_var;
+    new_var = &this->unk_283;
+    if ((this->unk_283 != 1) || func_8003305C(&this->actor, &this->unk_28C, globalCtx, 1)) {
+        if ((*new_var) != 0) {
+            this->unk_283 = (*new_var) | 2;
+        }
+        if (this->actor.bgCheckFlags & 2) {
+            this->actor.speedXZ = 0;
+        }
+        if (this->actor.bgCheckFlags & 1) {
+            if (this->actor.speedXZ < 0.0f) {
+                this->actor.speedXZ += 0.05f;
+            }
+        }
+
+        Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 1, 0x1194, 0);
+        if (SkelAnime_Update(&this->skelAnime) && (this->actor.bgCheckFlags & 1)) {
+            func_80AFCD60(this);
+        }
+    }
+}
+
+void func_80AFD7B4(EnSkb* this, GlobalContext* globalCtx) {
+    Animation_MorphToPlayOnce(&this->skelAnime, &D_060009DC, -4.0f);
+    this->actor.shape.rot.y = this->actor.yawTowardsPlayer;
+    this->actor.world.rot.y = this->actor.yawTowardsPlayer;
+    if (this->actor.bgCheckFlags & 1) {
+        this->actor.speedXZ = -6.0f;
+    }
+    this->unk_280 = 1;
+    this->actor.flags &= ~1;
+    func_80032E24(&this->unk_28C, 18, globalCtx);
+    this->unk_283 |= 4;
+    EffectSsDeadSound_SpawnStationary(globalCtx, &this->actor.projectedPos, NA_SE_EN_STALKID_DEAD, 1, 1, 0x28);
+    EnSkb_SetupAction(this, func_80AFD880);
+}
+
+void func_80AFD880(EnSkb* this, GlobalContext* globalCtx) {
+    if (func_8003305C(&this->actor, &this->unk_28C, globalCtx, 1) != 0) {
+        if (0.01f == this->actor.scale.x) {
```

Comment:
```suggestion
        if (this->actor.scale.x == 0.01f) {
```
Does the comparison need to go in that order?